cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -24,21 +24,25 @@ import (
 	"net/url"
 	"path/filepath"
 	"strings"
+	"time"
 
 	"github.com/google/go-github/v33/github"
 	"github.com/pkg/errors"
 	"golang.org/x/oauth2"
 	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/apimachinery/pkg/util/wait"
 
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
 )
 
 const (
-	httpsScheme              = "https"
-	githubDomain             = "github.com"
-	githubReleaseRepository  = "releases"
-	githubLatestReleaseLabel = "latest"
+	httpsScheme                = "https"
+	githubDomain               = "github.com"
+	githubReleaseRepository    = "releases"
+	githubLatestReleaseLabel   = "latest"
+	retryableOperationInterval = 3 * time.Second
+	retryableOperationTimeout  = 5 * time.Minute
```

Comment:
Given that GitHub is rate limited I suggest having a longer interval and shorter Timeout
Otherwise lgtm

Reply:
interval increased to every 10 seconds, timeout to 1 minute